src/NHibernate/Dialect/SybaseSQLAnywhere10Dialect.cs

```diff
@@ -556,6 +565,11 @@ public override SqlString GetLimitString(SqlString sql, SqlString offset, SqlStr
 
 			if (insertionPoint > 0)
 			{
+				if (limit == null && offset == null)
+					throw new ArgumentException("Cannot limit with neither a limit nor an offset");
+				if (limit == null)
+					throw new NotSupportedException($"Dialect {this} does not support setting an offset without a limit");
```

Comment:
This case was previously blowing with a null reference exception. Overriden in the 17 version since SQL Anywhere 12.0.1 (but not 12.0.0) has a syntax allowing the support of this case.